File: python/redispy/simpletls.py
import redis
import sys
from pathlib import Path


def main():
    if not (len(sys.argv) == 3 or len(sys.argv) == 4):
        print("Usage: host port password")
        exit(1)
    host = sys.argv[1]
    port = sys.argv[2]
    password = None
    if len(sys.argv) == 4:
        password = sys.argv[3]

    r = redis.StrictRedis(
        host=host,
        port=port,
        password=password,
        ssl=True,
        ssl_keyfile="../../testscripts/tls/db_key.pem",
        ssl_certfile="../../testscripts/tls/db_cert.pem",
        ssl_cert_reqs="required",
        ssl_ca_certs="../../testscripts/tls/ca_cert.pem",
    )
    print("Connected to Redis")

    print("Set: {}".format(r.set("foo", "bar")))
    print("Get: {}".format(r.get("foo")))

    r.connection_pool.disconnect()


if __name__ == "__main__":
    main()

File: node.js/ioredis/simpletls.js
'use strict';
if (!(process.argv.length === 4 || process.argv.length === 5)) {
    console.log("Usage: host port password")
    process.exit(1)
}
const Redis = require("ioredis");
const fs = require('fs');

var host = process.argv[2];
var port = process.argv[3];
var password = undefined;
if (process.argv.length === 5) {
    password = process.argv[4];
}
var tlsOpts = {
    key: fs.readFileSync('../../testscripts/tls/db_key.pem'),
    cert: fs.readFileSync('../../testscripts/tls/db_cert.pem'),
    ca: [fs.readFileSync('../../testscripts/tls/ca_cert.pem')],
    enableTrace: true,
    checkServerIdentity: (hostname, cert) => {return null;}, //return null if you do not want to verify hostname
};
var redis = new Redis({
    port: port,
    host: host,
    password: password,
    tls: tlsOpts
});
redis.on("error", function (error) {
    console.error("Error in connection");
    console.error(error);
});

redis.set("foo", "bar", function (err, result) {
    if (err) {
        return console.error(err);
    }
    console.log("SET:", result);
});

redis.get("foo").then(function (result) {
    console.log("GET:", result);
});
//close connection
// we wait for a second for all commands to complete and then close the connection,
setTimeout(function () { redis.quit() }, 1000);